reposerver/repository/repository.go

```diff
@@ -1456,6 +1460,8 @@ func newEnv(q *apiclient.ManifestRequest, revision string) *v1alpha1.Env {
 		&v1alpha1.EnvEntry{Name: "ARGOCD_APP_SOURCE_REPO_URL", Value: q.Repo.Repo},
 		&v1alpha1.EnvEntry{Name: "ARGOCD_APP_SOURCE_PATH", Value: q.ApplicationSource.Path},
 		&v1alpha1.EnvEntry{Name: "ARGOCD_APP_SOURCE_TARGET_REVISION", Value: q.ApplicationSource.TargetRevision},
+		&v1alpha1.EnvEntry{Name: "ARGOCD_APP_METADATA", Value: string(metadata)},
```

Comment:
Suggestion from @alexmt - this is potentially dangerous because metadata includes a lot of info. They're large enough they could break shell.

Reply:
Yeah, went ahead and changed the approach.

We already have `ARGOCD_APP_NAME` & `ARGOCD_APP_NAMESPACE`, instead of exporting everything in metadata just added extra support for `LABELS`.